packages/features/booking-audit/lib/repository/PrismaAuditActorRepository.ts

```diff
@@ -25,5 +25,87 @@ export class PrismaAuditActorRepository implements IAuditActorRepository {
 
         return actor;
     }
+
+    async createIfNotExistsUserActor(params: { userUuid: string }) {
+        return this.deps.prismaClient.auditActor.upsert({
+            where: { userUuid: params.userUuid },
+            create: {
+                type: "USER",
+                userUuid: params.userUuid,
+            },
+            update: {},
+        });
+    }
+
+    async createIfNotExistsGuestActor(email: string | null, name: string | null, phone: string | null) {
+        const normalizedEmail = email && email.trim() !== "" ? email : null;
+        const normalizedName = name && name.trim() !== "" ? name : null;
+        const normalizedPhone = phone && phone.trim() !== "" ? phone : null;
+
+        // If all fields are null, we can't use upsert (no unique constraint), so just create a new record
+        if (!normalizedEmail && !normalizedPhone) {
+            return this.deps.prismaClient.auditActor.create({
+                data: {
+                    type: "GUEST",
+                    email: null,
+                    name: normalizedName,
+                    phone: null,
+                },
+            });
+        }
+
+        // First try to find by email if email exists
+        if (normalizedEmail) {
```

Comment:
We should handle the scenario if an actor has both an email and phone values. Right now it'll make 4 possible DB calls because of the two `if` statements

Reply:
Let me think about how to handle it well.